src/hooks/useSyncChainQuery.ts

```diff
@@ -27,6 +27,7 @@ export default function useSyncChainQuery() {
 
   const urlChainId = getParsedChainId(parsedQs)
   const previousUrlChainId = usePrevious(urlChainId)
+  const [nextChainId, setNextChainId] = useState<number | undefined>(undefined)
```

Comment:
why do we have a current, next, and previous?

Reply:
sorta confusing but open to better variable names. `urlChainId` is what's in the query at the time. `previous` is what allows us to check if it changed. `next` is what we're saying it should update the actual chainId to when ready.